songpal/method.py

```diff
@@ -152,18 +155,63 @@ async def __call__(self, *args, **kwargs):
 
     @property
     def inputs(self) -> Dict[str, type]:
-        """Input parameters for this method."""
-        return self.signature.input
+        """Input parameters for this method version."""
+        return self.signatures[self._version].input or {}
+
+    @property
+    def latest_supported_version(self) -> Optional[str]:
+        """Latest version supported by this method."""
+        return (
+            max(self._supported_versions) if len(self._supported_versions) > 0 else None
```

Comment:
```suggestion
            max(self._supported_versions) if self._supported_versions else None
```
This is the last one, I promise! :-)

Reply:
All good, shouldn't have missed it :)

Reply:
I originally updated them all based on lint. It is nicer code, but I wasn't sure it was as human-readable. Probably more a coding ability thing :)